### PR TITLE
Issue 14140 - Bad codegen for CTFE union initialisers for immutable structs

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14140

### CODE REVIEW

**Comment by WalterBright** [comment]
Why is 9057 included in this PR?

**Comment by 9rnsr** [comment]
> Why is 9057 included in this PR?

See the commit message. 9057 is not really fixed, it's just hidden. The workaround code is still needed to keep the hiding.

File: src/aggregate.d [hunk 1]
```diff
@@ -280,6 +282,225 @@ public:
...
+                    errors = true;
+                }
+            }
+        }
+        return errors;
+    }
+
+    /***************************************
+     * Fill out remainder of elements[] with default initializers for fields[].
+     * Param:
```
Comment (WalterBright) [inline src/aggregate.d]: Params, not Param

File: src/aggregate.d [hunk 2]
```diff
@@ -280,6 +282,225 @@ public:
...
+     *      loc
+     *      elements    explicit arguments which given to construct object.
```
Comment (WalterBright) [inline src/aggregate.d]: `elements =`, not `elements`

File: src/aggregate.d [hunk 3]
```diff
@@ -280,6 +282,225 @@ public:
...
+     *      ctorinit    true if the elements will be used for default initialization.
```
Comment (WalterBright) [inline src/aggregate.d]: `ctorinit =`

File: src/aggregate.d [hunk 4]
```diff
@@ -280,6 +282,225 @@ public:
...
+                }
+            }
+        }
+        return errors;
+    }
+
+    /***************************************
+     * Fill out remainder of elements[] with default initializers for fields[].
+     * Param:
+     *      loc
```
Comment (WalterBright) [inline src/aggregate.d]: `loc = location`

File: src/dinterpret.d [hunk 5]
```diff
@@ -3646,6 +3673,22 @@ public:
         return;
     }
 
+    /* Set all sibling fields which overlapping with v to VoidExp.
```
Comment (WalterBright) [inline src/dinterpret.d]: /overlapping/overlap/

File: src/aggregate.d (hunk 1 above)
Comment (9rnsr) [inline src/aggregate.d]: Done.

File: src/aggregate.d (hunk 4 above)
Comment (9rnsr) [inline src/aggregate.d]: Done for this and other two parameters.

File: src/dinterpret.d (hunk 5 above)
Comment (9rnsr) [inline src/dinterpret.d]: Fixed.

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**